Arduino/micro-ros_runservo_service/micro-ros_runservo_service.ino

```diff
@@ -0,0 +1,103 @@
+#include <micro_ros_arduino.h>
+#include <trailbot_interfaces/srv/run_servo.h>
+#include <stdio.h>
+#include <rcl/error_handling.h>
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+#include <micro_ros_utilities/string_utilities.h>
+#include <Servo.h>
+
+Servo servo1, servo2, servo3, servo4;
+
+
+rcl_node_t node;
+rclc_support_t support;
+rcl_allocator_t allocator;
+rclc_executor_t executor;
+
+rcl_service_t service;
+rcl_wait_set_t wait_set;
+
+trailbot_interfaces__srv__RunServo_Response res;
+trailbot_interfaces__srv__RunServo_Request req;
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){while(1){};}}
+#define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){}}
+
+void service_callback(const void * req, void * res){
+  trailbot_interfaces__srv__RunServo_Request * req_in = (trailbot_interfaces__srv__RunServo_Request *) req;
+  trailbot_interfaces__srv__RunServo_Response * res_in = (trailbot_interfaces__srv__RunServo_Response *) res;
+
+  int32_t servoId = req_in->servo;
+  switch(servoId){
+    case 1:
+      servo1.writeMicroseconds(2000);
+      delay(950);
+      servo1.writeMicroseconds(1500);
+      delay(950);
+      res_in->success = true;
+      res_in->message = micro_ros_string_utilities_set(res_in->message, "Servo1 operation succeeded");   
+      break;
```

Comment:
The only things different from case to case are the servo ID, servo object, and response string. Can we extract this into a function to remove the duplicated code?

Reply:
servos may require different times as well to complete one cycle

Reply:
resolved